src/caffe/net.cpp

```diff
@@ -650,6 +650,61 @@ void Net<Dtype>::ShareTrainedLayersWith(const Net* other) {
 }
 
 template <typename Dtype>
+void Net<Dtype>::ShareBlobsWith(const Net* other) {
+  int num_source_blobs = other->blobs().size();
```

Comment:
Parameter blobs are shared by `ShareTrainedLayerWith` method. In the implementation of Caffe, `Solver` class calls it before testing the trained net in `Test()` method.

As far as I am understanding, the test nets could be partially different or even totally different from train net. At least, in almost cases, `DataLayer`s are different from train net.